src/coreclr/tools/aot/ILCompiler.ReadyToRun/Interop/IL/Marshaller.ReadyToRun.cs

```diff
@@ -25,12 +25,15 @@ protected static Marshaller CreateMarshaller(MarshallerKind kind)
                     return new BooleanMarshaller();
                 case MarshallerKind.AnsiString:
                     return new AnsiStringMarshaller();
+#if !READYTORUN
+                // UTF8 and Unicode Strings are not supported in R2R
                 case MarshallerKind.UTF8String:
                     return new UTF8StringMarshaller();
-                case MarshallerKind.SafeHandle:
-                    return new SafeHandleMarshaller();
                 case MarshallerKind.UnicodeString:
                     return new UnicodeStringMarshaller();
```

Comment:
Unicode marshaller is very frequently used on Windows, so removing it here will likely cause more methods to be JITed during startup.

Reply:
yeah Michal also pointed that out above, will need to handle the non-supported cases so as to precompile the most common cases.